internal/sourceanalysis/sourceanalysis.go

```diff
@@ -0,0 +1,84 @@
+package sourceanalysis
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"golang.org/x/exp/slices"
+	"golang.org/x/vuln/exp/govulncheck"
+
+	"github.com/google/osv-scanner/internal/govulncheckshim"
+	"github.com/google/osv-scanner/internal/output"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+// vulnsFromAllPkgs returns the flattened list of unique vulnerabilities
+func vulnsFromAllPkgs(pkgs []models.PackageVulns) []models.Vulnerability {
+	flatVulns := map[string]models.Vulnerability{}
+	for _, pv := range pkgs {
+		for _, vuln := range pv.Vulnerabilities {
+			flatVulns[vuln.ID] = vuln
+		}
+	}
+
+	vulnList := []models.Vulnerability{}
+	for _, v := range flatVulns {
+		vulnList = append(vulnList, v)
+	}
+
+	return vulnList
+}
+
+// Run runs the language specific analyzers on the code given packages and source info
+func Run(r *output.Reporter, source models.SourceInfo, pkgs []models.PackageVulns) {
+	vulnsSlice := vulnsFromAllPkgs(pkgs)
```

Comment:
nit: let's just name this `vulns` (to be consistent with the above function too).